tests/neo4j/test_summary.py

```diff
@@ -0,0 +1,234 @@
+    def test_can_obtain_notification_info(self):
+        summary = self.get_summary("EXPLAIN MATCH (n), (m) RETURN n, m")
+        notifications = summary.notifications
+        self.assertIsInstance(notifications, list)
+        self.assertEqual(len(notifications), 1)
+        self.assertIsInstance(notifications[0], dict)
+
+    @driver_feature(types.Feature.TMP_FULL_SUMMARY)
+    def test_contains_time_information(self):
+        summary = self.get_summary("UNWIND range(1, 100) AS n "
+                                   "RETURN n AS number")
+
+        self.assertIsInstance(summary.result_available_after, int)
+        self.assertIsInstance(summary.result_consumed_after, int)
+
+        self.assertGreaterEqual(summary.result_available_after, 0)
+        self.assertGreaterEqual(summary.result_consumed_after, 0)
+
+    @driver_feature(types.Feature.TMP_FULL_SUMMARY)
+    def test_protocol_version_information(self):
+        summary = self.get_summary("RETURN 1 AS number")
+
+        self.assertEqual(summary.server_info.protocol_version,
+                         get_server_info().max_protocol_version)
+
+    def test_agent_string(self):
+        summary = self.get_summary("RETURN 1 AS number")
+        if isinstance(summary, dict) and get_driver_name() in ["java"]:
+            self.skipTest("Java 4.2 backend does not support summary")
+
+        self.assertTrue(summary.server_info.agent.startswith("Neo4j/"))
+        version = summary.server_info.agent[6:].split(".")
+        self.assertEqual(version[:2], get_server_info().version.split("."))
+
+    def _assert_counters(self, summary, nodes_created=0, nodes_deleted=0,
+                         relationships_created=0, relationships_deleted=0,
+                         properties_set=0, labels_added=0, labels_removed=0,
+                         indexes_added=0, indexes_removed=0,
+                         constraints_added=0, constraints_removed=0,
+                         system_updates=0, contains_updates=False,
+                         contains_system_updates=False):
+        for attr, val in locals().items():
+            if attr in ("self", "summary"):
+                continue
+            counter_val = getattr(summary.counters, attr)
+            self.assertIsInstance(counter_val, type(val))
+            self.assertEqual(counter_val, val)
+
+    @driver_feature(types.Feature.TMP_FULL_SUMMARY)
+    def test_summary_counters_case_1(self):
+        params = {"number": types.CypherInt(3)}
+
+        summary = self.get_summary("RETURN $number AS x", params=params)
+
+        self.assertEqual(summary.query, "RETURN $number AS x")
+        self.assertEqual(summary.parameters, params)
+
+        self.assertIn(summary.query_type, ("r", "w", "rw", "s"))
+
+        self._assert_counters(summary)
+
+    @driver_feature(types.Feature.TMP_RESULT_KEYS,
+                    types.Feature.TMP_FULL_SUMMARY)
+    def test_summary_counters_case_2(self):
```

Comment:
I assume it is a ported version of an existing test. In the future, it may be worth splitting it up.

Reply:
Yes, the tests weren't particularly well named or structured. Maybe I should just do the splitting up right now, before actually merging in this tech debt.